src/Apps/Order2/Routes/Checkout/Components/CriticalErrorModal.tsx

```diff
@@ -0,0 +1,61 @@
+import { Button, Flex, ModalDialog, Text } from "@artsy/palette"
+import { useReturnToArtwork } from "Apps/Order2/Routes/Checkout/Hooks/useReturnToArtwork"
+
+export const CriticalErrorModal: React.FC<{
+  error: string | null
+}> = ({ error }) => {
+  const returnToArtwork = useReturnToArtwork()
+
+  if (!error) {
+    return null
+  }
+
+  const handleClose = () => {
+    returnToArtwork({ force: true })
+  }
+
+  const handleReload = () => {
+    // Remove all event listeners before reloading to avoid confirmation dialog
+    window.onbeforeunload = null
+    window.location.reload()
+  }
```

Comment:
I think this `window` code should be located somewhere similar to how it's hidden inside `returnToArtwork`. Or we could just do it from this component.